File: base/src/test/java/io/quarkus/code/rest/QuarkusProjectServiceMock.java
package io.quarkus.code.rest;

import io.quarkus.code.service.QuarkusProjectService;
import io.quarkus.code.model.ProjectDefinition;
import io.quarkus.code.service.PlatformInfo;
import io.quarkus.devtools.commands.data.QuarkusCommandException;
import io.quarkus.test.Mock;

import java.io.IOException;
import java.util.concurrent.atomic.AtomicReference;

import jakarta.inject.Singleton;

@Mock
@Singleton
public class QuarkusProjectServiceMock extends QuarkusProjectService {

    private AtomicReference<ProjectDefinition> createdProjectRef = new AtomicReference<>();

    @Override
    public byte[] create(PlatformInfo platformInfo, ProjectDefinition projectDefinition)
            throws IOException, QuarkusCommandException {
        createdProjectRef.set(projectDefinition);
        return super.create(platformInfo, projectDefinition);
    }

    public ProjectDefinition getCreatedProject() {
        return this.createdProjectRef.get();
    }

}

File: base/src/test/kotlin/io/quarkus/code/rest/QuarkusProjectServiceMock.kt
package io.quarkus.code.rest

import io.quarkus.code.service.QuarkusProjectService
import io.quarkus.code.model.ProjectDefinition
import io.quarkus.code.service.PlatformInfo
import io.quarkus.test.Mock
import java.util.concurrent.atomic.AtomicReference
import jakarta.inject.Singleton

@Mock
@Singleton
open class QuarkusProjectServiceMock: QuarkusProjectService() {

    val createdProjectRef : AtomicReference<ProjectDefinition> = AtomicReference()

    override fun create(platformInfo: PlatformInfo, projectDefinition: ProjectDefinition): ByteArray {
        createdProjectRef.set(projectDefinition)
        return super.create(platformInfo, projectDefinition)
    }

    fun getCreatedProject(): ProjectDefinition {
        return this.createdProjectRef.get()
    }

}